community/kernel/src/main/java/org/neo4j/kernel/impl/locking/LockClientStateHolder.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.locking;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * State control class for Locks.Clients.
+ * Client state represent current Locks.Client state: OPEN/CLOSED
+ * and number of active clients
+ */
+public class LockClientStateHolder
```

Comment:
* same state holder class will be used as well in forseti client for same purpose (separate PR), thats why it's not private class.
* yes good idea, will make it final
* do not really agree that it should be subclass of `AtomicInteger because it does not really have the same purpose and should have a bit more narrow API comparing to `AtomicInteger`